### PR TITLE
Testing with Rspec

### PR DESCRIPTION
**Pull request by eypsrcnuygr**
*There are 20 test cases.
*The tests have been made by using 4 arrays, 1 proc, and some inline blocks.
*To run tests you need to install RSpec and then run 'rspec'.

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    it 'returns the elements of the array' do
+      checker = array.my_each { |x| x }
+      expect(checker).to eq(array)
+    end
+  end
+  describe '#my_each_with_index' do
+    it 'returns an enumerator when no block_given' do
+      expect(array.my_each).to be_instance_of(Enumerator)
+    end
+    it 'returns the sum of array ' do
```
Comment (kubilaycaglayan) [inline spec/enumerable_spec.rb:19]: You are checking "my_each" in the "my_each_with_index" block.

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      sum = 0
+      array.my_each_with_index { |item| sum += item }
+      expect(sum).to eql(43)
+    end
+  end
+  describe '#my_select' do
+    it 'returns numbers > 5' do
+      expect(array.my_select { |x| x > 5 }).to eql([7, 9, 8, 10])
+    end
+  end
```
Comment (kubilaycaglayan) [inline spec/enumerable_spec.rb:29]: Negative scenario could be written. E.g. < 5

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    it 'returns true if all items are true' do
+      expect(array.my_all? { |el| el > 0 }).to eq(true)
+    end
+  end
+
+  describe '#my_any?' do
+    it 'returns true if one element is true' do
+      checker = array.my_any? { |item| item > 5 }
+      expect(checker).to eql(true)
+    end
```
Comment (kubilaycaglayan) [inline spec/enumerable_spec.rb:40]: Negative scenario could be written. E.g. returns false

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  end
+
+  describe '#my_none?' do
+    it 'returns true if all elements return false' do
+      checker = array.my_none? { |item| item > 15 }
+      expect(checker).to eql(true)
+    end
```
Comment (kubilaycaglayan) [inline spec/enumerable_spec.rb:47]: Negative scenario could be written. E.g. returns false if one of them true.
**And also please check the rest of the code to add both negative and positive scenarios.** :pray:

File: spec/enumerable_spec.rb (hunk 1 above)
Comment (eypsrcnuygr) [reply spec/enumerable_spec.rb:19]: It's fixed.

File: spec/enumerable_spec.rb (hunk 2 above)
Comment (eypsrcnuygr) [reply spec/enumerable_spec.rb:29]: It's added as a negative case when no element has passed the test case.

File: spec/enumerable_spec.rb (hunk 3 above)
Comment (eypsrcnuygr) [reply spec/enumerable_spec.rb:47]: A negative case has been added.

File: spec/enumerable_spec.rb (hunk 4 above)
Comment (eypsrcnuygr) [reply spec/enumerable_spec.rb:54]: The negative scenarios have been added to my_none?, my_any?, my_all? and my_select.

**Comment by eypsrcnuygr** [comment]
*There are 16 example tests.
*All tests have been made by using the 4 different types of arrays that have been added at top of the Enumerable describe method.
*Negative cases have been added to my_none?, my_any?, my_all? and my_select.

File: spec/enumerable_spec.rb [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      end
+    end
+  end
+
+  describe '#my_map' do
+    it 'returns a new array with block or proc called' do
+      checker = array.my_map { |x| x * 2 }
+      expect(checker).to eql [2, 6, 10, 14, 18, 16, 20]
+    end
+  end
```
Comment (bertil291utn) [inline spec/enumerable_spec.rb:83]: ## More cases
- [ ] Please add more cases just once it's not enough.

*Example:*

```ruby
describe '#my_map' do
    it 'when no block is given, returns an enumerator' do
      expect(array_numbers.my_map).to be_an Enumerator
    end
    context 'given block/proc' do
      it 'returns same array as original with given block' do
        expect(array_numbers.my_map { |elem| elem * 10 }).to eq([100, 20, 30, 40, 10, 50])
      end

      it 'returns same array as original with given proc' do
        expect(array_numbers.my_map(&my_proc_map)).to eq([100, 20, 30, 40, 10, 50])
      end
    end
  end
```

- [ ] Also negative cases

File: spec/enumerable_spec.rb [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+  describe '#my_inject' do
+    it 'return one element that is the result for the process' do
+      checker = array.my_inject(5) { |x, y| x + y }
+      expect(checker).to eql(48)
+    end
+  end
```
Comment (bertil291utn) [inline spec/enumerable_spec.rb:90]: ## Negative cases

- [ ] Please add negative cases for this method

**Comment by eypsrcnuygr** [comment]
*Negative cases added to the my_map and my_inject.
*20 test cases in total.
*All tests have been passed.
*The tests are done by using 4 arrays and one proc and some inline blocks.

**Pull request merged**